### PR TITLE
[RFR] JSON field

### LINKED ISSUE #224: json column type
Not for all users, but very handy nonetheless... Should only serialize/deserialize on top of a text field.

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Add JSON field
- [x] Validate JSON format before submission
- [x] add JSON column
- [x] Add JSON formatter to improve edition usability

Show view:
image

Edit view:
image

Edit view error:
image

Closes #224

### CODE REVIEW

**Comment by fzaninotto** [comment]
AWESOME :clap:

File: src/javascripts/ng-admin/Main/MainModule.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ define(function (require) {
 
     MainModule.directive('maDashboardPanel', require('ng-admin/Main/component/directive/maDashboardPanel'));
     MainModule.directive('menu', require('ng-admin/Main/component/directive/Menu'));
+    MainModule.directive('maJsonValidator', require('ng-admin/Main/component/directive/maJsonValidator'));
```
Comment (fzaninotto) [inline src/javascripts/ng-admin/Main/MainModule.js]: Why is it in main and not in crud ?

File: src/javascripts/test/unit/Crud/field/maJsonFieldSpec.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            expect(element.find('textarea').val()).toBe(JSON.stringify(JSON.parse('["foo",{"bar":123}]'), null, 2));
+        });
+
+        it("should convert the JSON string into a JavaScript object value", function () {
+            scope.field = new Field();
+            var element = $compile(directiveUsage)(scope);
+            scope.$digest();
+            element.find('textarea').val(JSON.stringify(JSON.parse('["foo",{"bar":456}]'), null, 2));
+            // ngModel listens for "input" event
+            // http://stackoverflow.com/questions/15739960/how-does-angularjs-internally-catch-events-like-onclick-onchange/15740287#15740287
```
Comment (fzaninotto) [inline src/javascripts/test/unit/Crud/field/maJsonFieldSpec.js]: Good catch, I was blocked on this one for hours!

**Comment by endel** [comment]
That's pretty. Well done, guys.

File: src/javascripts/ng-admin/Main/MainModule.js (hunk 1 above)
Comment (jeromemacias) [inline src/javascripts/ng-admin/Main/MainModule.js]: I really don't knew where to put this file, is `Crud/field/maJsonValidator.js` sounds better for you ?

Comment (fzaninotto) [inline src/javascripts/ng-admin/Main/MainModule.js]: Why not under crud/validator/maJsonValidator ?

Comment (fzaninotto) [inline src/javascripts/ng-admin/Main/MainModule.js]: In fact why isn't it part of the main `maJsonField` directive?

**Pull request merged**